docs: Improve contributing (#1303)
probabl-ai/skore: CONTRIBUTING.rst

Examples of correct PR titles: ``docs: Update the docstrings`` or ``feat: Remove CrossValidationAggregationItem.``

Tests
-----

To run the tests locally, you may run

.. code-block:: bash

    make test

To run only back-end tests, use:

.. code-block:: bash

>     make test-back-end

Reviewer: ```suggestion
    make test-backend
```

    open build/html/index.html

> The PR will also build the documentation and a bot will indicate the link to the documentation preview to easily check the results.

Reviewer: ```suggestion
The PR will also build the documentation and a bot will automatically add a comment with a link to the documentation preview to easily check the results.
```

#. Text is written in US english ("visualize" rather than "visualise")
#. In the CLI, positional arguments are written in snake case (``snake_case``), keyword arguments in kebab case (``kebab-case``)
#. When there is a default argument, it should be shown in the help message, typically with ``(default: <default value>)`` at the end of the message

Contributing to the examples
----------------------------

The examples are stored in the folder called `examples`. They are classified in subcategories.

> They should be written in a python file, with cells marked by `# %%`. They will be automatically transcripted to rst files in the subfolder `sphinx/auto_examples`. This subfolder is listed in the gitignore, and cannot be pushed.

Reviewer: ```suggestion
They should be written in a python file, with cells marked by `# %%`. They will be automatically transcripted to rst files in the subfolder `sphinx/auto_examples`. This subfolder is listed in the gitignore, and cannot be pushed.
```
```suggestion
They should be written in a python file, with cells marked by `# %%`. They will be automatically converted to rst files in the subfolder `sphinx/auto_examples`. This subfolder is listed in the gitignore, and cannot be pushed.
```

The python file should start by a docstring indicating the example name and the title.

No example should require to have large files in this repository. For example, no dataset should be stored, it should be downloaded in the script.

Contributing to the ReadMe
--------------------------

The README.md file can be modified and is part of the documentation.

> This file is used to be presented on [Pypi](https://pypi.org/project/skore/#description).

Reviewer: ```suggestion
This file is used to be presented on [PyPI](https://pypi.org/project/skore/#description).
```

Reviewer: Similarly the programming language is usually written `Python` (capital P), but either way is fine
